### DISCUSSION TITLE
feature request: select options with ulauncher, like dmenu

### DISCUSSION THREAD

**Discussion by rodrigoGA** [Ulauncher/Ulauncher#898] [Ideas]
## Question
I want to know if it is possible to select a folder with ulauncher. Something similar to what dmenu, rofi, or fzf does

command example with rofi

    ls ~/Projects | rofi -show -dmenu -i | xargs -I_ sh ~/Projects/_/open.sh

List the projects folder, select one, and then run open.sh. I have this command in a keyboard shortcut.

Is it possible to do this with ulauncher?

**Comment by friday** [comment]
It's not possible yet at least. Some other suggestions here would be able to do something similar if implemented.

**Comment by friday** [comment]
I tried searching for an issue and found none but I know there's been suggestions to trigger ulauncher-toggle with a text argument, which would be able to do something like this in combination with an extension.

Ex `ulauncher-toggle myextensionprefix /path/to/folder`

Edit: I found it: #312 It only covers triggering the keyword/extension though. This feature would also need to pass an argument, so it may not be fully covered by that issue.

**Comment by rodrigoGA** [comment]
if this feature were added it would enable many functionalities. I think it would be cool
It is something very simple and very flexible, it can be used for many functionalities, from opening a project, connecting to a vpn, selecting an emoticon ... or whatever you want.

**Comment by friday** [comment]
I also think it would be cool, but it's very much outside of the scope of what Ulauncher does today, the target audience Ulauncher caters to, and it's not clear how to implement it. Ulauncher runs as a background service and ulauncher-toggle takes no arguments. Ulauncher always shows icons, and normally both a name and a description, but this way all it would do is get a list of strings. It wouldn't know it they were file system paths, and if it were to know we would have to have some kind of system in place to detect that these are paths, VPN connections, emoticons, or whatever. dmenu was designed for this, but Ulauncher wasn't.

My earlier suggestion would make it possible to pass a CLI argument to an extension, but that's about as far as I think this is possible. The extension would be in control of the "action" taken given the input, and you would see the CLI argument in your query.
